<a href="https://colab.research.google.com/github/samantafe8-collab/Ciencia-datos-curso/blob/main/tarea_02_pandas_ingestion.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Tarea 02 — Carga de Datos con Pandas

**Instrucciones:** Completa cada celda de código con tu respuesta.  
Usarás el dataset de Iris disponible en la URL indicada.

In [1]:
import pandas as pd

---
## Parte 1 — Cargar y explorar datos

**P1.** Carga el dataset de Iris desde la siguiente URL y guárdalo en una variable llamada `df`.  
URL: `https://raw.githubusercontent.com/mwaskom/seaborn-data/master/iris.csv`

In [2]:
url = "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/iris.csv"

# Carga el data frame

df = pd.read_csv(url)


**P2.** Muestra las primeras 5 filas del DataFrame.

In [3]:
# Tu respuesta aquí

df.head()

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


**P3.** Usa `.info()` para ver los tipos de columnas y si hay valores nulos.  
Luego usa `.describe()` para ver el resumen estadístico. ¿Cuántas filas tiene el dataset?

In [4]:
# Tu respuesta aquí
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


**Respuesta:** El dataset tiene 150 filas, 4 de las 5 columnas son decimales y 1 columna es texto

---
## Parte 2 — Selección y filtrado

**P4.** Selecciona solo las columnas `sepal_length` y `species` e imprime las primeras 5 filas.

In [5]:
# Tu respuesta aquí
df[['sepal_length', 'species']].head()

,sepal_length,species
0,5.1,setosa
1,4.9,setosa
2,4.7,setosa
3,4.6,setosa
4,5.0,setosa


**P5.** Filtra el DataFrame para quedarte solo con las flores de la especie `virginica`.  
¿Cuántas filas tiene ese subconjunto?

In [6]:
# Tu respuesta aquí
df_virginica = df[df['species'] == 'virginica']

cantidad_filas = len(df_virginica)

print(f"El subconjunto de especie virginica tiene {cantidad_filas} filas.")

El subconjunto de especie virginica tiene 50 filas.


\*El subconjunto de especies virginica tiene 50 filas*

**P6.** Filtra las filas donde `petal_length` sea mayor a 5.0. Imprime cuántas filas cumplen esa condición.

In [7]:
# Tu respuesta aquí
df_petal_may_5 = df[df['petal_length'] > 5.0]

cant_filas_grandes = len(df_petal_may_5)

print(f"Hay {cant_filas_grandes} flores con pétalos mayores a 5.0.")

Hay 42 flores con pétalos mayores a 5.0.


*Hay 42 flores con pétalos mayores a 5*

---
## Parte 3 — Guardar datos

**P7.** Guarda el subconjunto de `virginica` (de P5) en un archivo CSV llamado `virginica.csv` en la carpeta `output/`.  
No incluyas el índice en el archivo guardado.

In [11]:
import os
os.makedirs("output", exist_ok=True)

df_virginica = df[df['species'] == 'virginica']

df_virginica.to_csv('output/virginica.csv', index=False)

print("El archivo se ha guardado temporalmente en la carpeta output")


El archivo se ha guardado temporalmente en la carpeta output


*El archivo csv se guardó temporalmente en la carpeta output*

**P8.** Carga de nuevo el archivo `output/virginica.csv` y verifica que tiene el número correcto de filas usando `.shape`.

In [12]:
# Tu respuesta aquí

# Cargar el archivo guardado
df_verificacion = pd.read_csv('output/virginica.csv')

# Verificar el número de filas y columnas
print(df_verificacion.shape)

(50, 5)


*Se verifica que el subconjunto de especies virginica tiene 50 filas*